core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -87,6 +104,11 @@ public void groupBy(StructuredRecord record, Emitter<String> emitter) throws Exc
         throw new IllegalArgumentException(
           String.format("Keyfield '%s' does not exist in input schema %s", keyField, record.getSchema()));
       }
+      StructuredRecord.Builder droppedRecordBuilder = StructuredRecord.builder(errorRecordSchema);
```

Comment:
Tested this with multiple mappers, it seems to be working fine and not overriding the keys.
However, I agree that in case of with multiple reducers, these keys could be overridden.

Can we keep the record itself as a key or do you have any other suggestion that we can use to maintain the unique key.